src/NuGet/Microsoft.Net.Compilers.Toolset/AnyCpu/build/Microsoft.Net.Compilers.Toolset.props

```diff
@@ -3,6 +3,7 @@
 
   <PropertyGroup>
     <_RoslynTargetDirectoryName Condition="'$(MSBuildRuntimeType)' == 'Core'">net6.0</_RoslynTargetDirectoryName>   
+    <_RoslynTargetDirectoryName Condition="'$(MSBuildRuntimeType)' == 'Core' AND '$(DotNetBuildFromSource)' != 'true'">net7.0</_RoslynTargetDirectoryName>   
```

Comment:
This does not seem correct. This is the props file that will exist in our NuPkg. That means it's evaluated on the customer machine, not on the build machine. On the customer machine `$(DotNetBuildFromSource)` won't be a variable.